src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/routinghelper/HorseTransportMode.java

```diff
@@ -0,0 +1,75 @@
+package org.openstreetmap.josm.plugins.pt_assistant.actions.routinghelper;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.drew.lang.annotations.NotNull;
+import org.openstreetmap.josm.data.osm.IRelation;
+import org.openstreetmap.josm.data.osm.IWay;
+import org.openstreetmap.josm.data.osm.Node;
+import org.openstreetmap.josm.data.osm.OsmPrimitiveType;
+import org.openstreetmap.josm.data.osm.Relation;
+import org.openstreetmap.josm.data.osm.Way;
+import org.openstreetmap.josm.plugins.pt_assistant.utils.PTIcons;
+import org.openstreetmap.josm.tools.I18n;
+import org.openstreetmap.josm.tools.ImageProvider;
+
+public class HorseTransportMode implements ITransportMode {
+    @Override
+    public boolean canTraverseWay(@NotNull final IWay<?> way, @NotNull final WayTraversalDirection direction) {
+        final String onewayValue = way.get("oneway");
+        List<String> majorHighways = Arrays.asList(
+            "tertiary", "secondary", "primary", "trunk");
+        majorHighways.forEach(v -> majorHighways.add(String.format("%s_link", v)));
+        // This list is ordered from most suitable to least suitable
+        List<String> suitableHighwaysForHorseRiders = Arrays.asList(
+            "bridleway", "pedestrian", "footway", "path", "track", "living_street", "residential", "unclassified", "cyclestreet", "service", "cycleway");
+        suitableHighwaysForHorseRiders.addAll(majorHighways); // TODO do this only once when plugin starts
```

Comment:
```suggestion
        final List<String> suitableHighwaysForHorseRiders = Stream.concat(
            // This list is ordered from most suitable to least suitable
            Stream.of("bridleway", "pedestrian", "footway", "path", "track", "living_street", "residential", "unclassified", "cyclestreet", "service", "cycleway"),
            Stream.of("tertiary", "secondary", "primary", "trunk").flatMap(it -> Stream.of(it, it + "_link"))
        ).collect(Collectors.toList());
```
This would be the "streamy" way of doing this :wink:
You can make this a (static) field of the `HorseTransportMode`, so it is initialized only once.